### DISCUSSION TITLE
Can not access the spreadsheet created

### DISCUSSION THREAD

**Discussion by brand17** [mobizt/ESP-Google-Sheet-Client#11] [Q&A]
Hi, I've created the spreadsheet with your example. As a result I've got a response with spreadsheetUrl. But when I try to access it - I am getting "Permission denied". As far as I understand - the file is created on service account and it should give me access to the file (https://stackoverflow.com/questions/37967937/google-sheets-api-v4-create-sheet-and-invite-users). How can I do it?

**Comment by mobizt** [comment]
It is already stated clearly in the comment
https://github.com/mobizt/ESP-Google-Sheet-Client/blob/d5ded41158df5250362cf80ffbef9b1f75432a5e/examples/Spreadsheets/Create/Create.ino#L33-L34

https://github.com/mobizt/ESP-Google-Sheet-Client/blob/d5ded41158df5250362cf80ffbef9b1f75432a5e/examples/Spreadsheets/Create/Create.ino#L104

**Reply by brand17** [reply]
It doesn't work: I've added my email (my Google account) there but still getting "Permission denied". I tried "Request access" but I didn't get any e-mail. I am able to update the created spreadsheet with your "update" example (even when signed out from google account). But not able to access the spreadsheet by URL from browser (Chrome).

**Reply by mobizt** [reply]
This is the Google service issue not a REST API client issue.

Google may limit your account, service account or your project by some reason that I don't know.

May be security concerns or payment issue.

You should create new project, create Service Account for it and set share the spreadsheet to your email again.

**Reply by mobizt** [reply]
Who creates (own) the spreadsheet? You (gmail user account) or Client (Service Account)

> I am able to update the created spreadsheet with your "update" example (even when signed out from google account).

If you own the spreadsheet and set share the access to the service account, client will be able to read and edit your spreadsheet but can't delete.

User (you) log in and log out are not related to your document you shared. Client is still able to read/edit your spreadsheet as usual.

If the spreadsheet creates by client (using service account), it requires user (you) email to set share. Then you will be able to open the spreadsheet url issued by calling the create function and can read and edit.

**Reply by brand17** [reply]
I have created the spreadsheet with "Create" example, so I believe Service account owns the spreadsheet. Do I need to pay for it? Because I didn't - I just created project and Service Account.

**Comment by mobizt** [comment]
The "permission denied" error is due to your service account does not have the permission to access the service resource, you need to read this and follow.
https://github.com/mobizt/ESP-Google-Sheet-Client#prerequisites

**Comment by brand17** [comment]
I've found my mistake - I didn't enable Google Drive API. Now all working. Thanks)
